week-6/Homework/extra/2-slideshow/slideshow.js

```diff
@@ -1 +1,61 @@
 // Write your code here
+const images = [
+  {
+    src:
+      "https://images.unsplash.com/photo-1524995997946-a1c2e315a42f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80",
+  },
+  {
+    src:
+      "https://images.unsplash.com/photo-1520387294843-fd994fd872e7?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=675&q=80",
+  },
+  {
+    src:
+      "https://images.unsplash.com/photo-1481627834876-b7833e8f5570?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=841&q=80",
+  },
+];
+
+let img = document.querySelector("img");
+let forBtn = document.querySelector("#forward");
+forBtn.addEventListener("click", slideForward);
+let backBtn = document.querySelector("#back");
+backBtn.addEventListener("click", slideBack);
+let autoForwardBtn = document.querySelector("#autoForward");
+autoForwardBtn.addEventListener("click", autoSlideForward);
+let autoBackBtn = document.querySelector("#autoBack");
+autoBackBtn.addEventListener("click", autoSlideBack);
+let stopBtn = document.querySelector("#stop");
+stopBtn.addEventListener("click", stopAutoPlay)
+let counter = 0;
+let interval;
+
+function slideForward() {
+  img.src = images[counter % images.length].src;
+  counter++;
+  console.log(counter % images.length);
+}
```

Comment:
It isn't ideal if your `slideForward()` function relies on continuing to increase the value of counter like this, since eventually this would cause an exception if the number got to be so large that it exceeded the allowed size of an integer.

You could avoid this problem by doing something like this, similar to how you've dealt with it in `slideBack()`, except you need to do it *before* accessing the `img.src` property:

```js
if (counter === images.length - 1) {
    counter = 0;
} else {
    counter++;
}
```

If you use if-else blocks like these to maintain `counter` then you also don't need to do `counter % images.length` as this value will be the same as `counter`.